### PR TITLE
pass option sourceMap to sveltePreprocess

### PR DESCRIPTION
**Pull request by milahu**
part of issue https://github.com/sveltejs/svelte-preprocess/issues/286

### CODE REVIEW

File: scripts/setupTypeScript.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ rollupConfig = rollupConfig.replace(`'src/main.js'`, `'src/main.ts'`)
 // Add preprocessor
 rollupConfig = rollupConfig.replace(
   'compilerOptions:',
-  'preprocess: sveltePreprocess(),\n\t\t\tcompilerOptions:'
+  'preprocess: sveltePreprocess({ sourceMap: !production }),\n\t\t\tcompilerOptions:'
```
Comment (benmccann) [inline scripts/setupTypeScript.js:68]: I wonder if we can have `sveltePreprocess` use a default value based off `process.env.NODE_ENV`? I could imagine people adding `sveltePreprocess` without this script (e.g. to a plain JS project, sapper project, webpack project, etc.) and getting this setting wrong

Comment (milahu) [reply scripts/setupTypeScript.js:68]: yeah but ... out of scope for this patch?

dev-mode is hard to detect cos different build tools create different `process.env`

diff of `JSON.stringify(process.env, null, 2)`

```diff
$ diff -u0 env.dev.txt env.build.txt
--- env.dev.txt 2021-01-26 18:25:02.041922665 +0100
+++ env.build.txt       2021-01-26 18:25:28.579592592 +0100
@@ -36 +36 @@
-  "npm_lifecycle_event": "dev",
+  "npm_lifecycle_event": "build",
@@ -48 +48 @@
-  "npm_config_argv": "{\"remain\":[\"run\",\"dev\"],\"cooked\":[\"run\",\"dev\"],\"original\":[\"run\",\"dev\"]}",
+  "npm_config_argv": "{\"remain\":[\"run\",\"build\"],\"cooked\":[\"run\",\"build\"],\"original\":[\"run\",\"build\"]}",
@@ -65 +65 @@
-  "npm_lifecycle_script": "rollup -c -w",
+  "npm_lifecycle_script": "rollup -c",
@@ -114,2 +114 @@
-  "npm_package_devDependencies_svelte_check": "^1.1.29",
-  "ROLLUP_WATCH": "true"
+  "npm_package_devDependencies_svelte_check": "^1.1.29"
```

to reduce redundancy, i would prefer a global `sourceMap` option for `rollup-plugin-svelte`

Comment (benmccann) [reply scripts/setupTypeScript.js:68]: Ok. Makes sense. In that case, we should also update https://github.com/sveltejs/sapper-template and https://github.com/sveltejs/template-webpack/

**Review by dummdidumm — APPROVED** [review]

**Comment by milahu** [comment]
> we should also update https://github.com/sveltejs/sapper-template and https://github.com/sveltejs/template-webpack/

* https://github.com/sveltejs/sapper-template/pull/299
* https://github.com/sveltejs/template-webpack/pull/37

**Comment by benmccann** [comment]
Thank you!

**Pull request merged**